avm/res/network/nat-gateway/main.bicep

```
@@ -5,6 +5,14 @@ metadata owner = 'Azure/module-maintainers'
 @description('Required. Name of the Azure Bastion resource.')
 param name string
 
+@description('Required. A list of availability zones denoting the zone in which Nat Gateway should be deployed.')
+@allowed([
+  1
+  2
+  3
+])
+param zones array
```

If it can only be one, it should be 
```suggestion
param zones int
```

or even
```suggestion
param availabilityZone int
```
like in the Compute VirtualMachine module

@ChrisSidebotham / @eriqua Though I'm not 100% sure about the rename of `zones` to `availabilityZone` It's a lot more descriptive (singular + more meaningful) and makes much more sense (and is implemented like this in the VM module), yet it's also a different name than the API.

---

Also, please note, if you make it an int, I'd recommend to update the allowedSet and description to what is used in the VM module to enable one to 'not' set it explicitely using a value of `0`

---

Unresolving this comment as there were a few more points to it ;) Also reached out to the team to agree on an approach here :)

---

Alright @fabmas,
just got word. Please implement the parameter as follows:
```bicep
@description('Required. If set to 1, 2 or 3, the Availability Zone is set to that value. If 0, then Availability Zones is not used.')
@allowed([
  0
  1
  2
  3
])
param zone int
``` 

alongside the implementation
```bicep
resource natGateway (...) {
    (...)
  properties: {
    (...)
    zones: zone != 0 ? [ string(zone) ]  : null
  }
}

module publicIPAddresses 'br/public:avm/res/network/public-ip-address:0.2.1' = [
  for (publicIPAddressObject, index) in (publicIPAddressObjects ?? []): {
    name: '${uniqueString(deployment().name, location)}-NatGw-PIP-${index}'
    params: {
      (...)
      zones: publicIPAddressObject.?zones ?? (availabilityZone != 0 ? [ string(zone) ]  : null)
    }
 }
```